pkg/validation/internal/multiarch.go

```diff
@@ -0,0 +1,623 @@
+package internal
+
+import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"sort"
+	"strings"
+
+	"github.com/operator-framework/api/pkg/manifests"
+	"github.com/operator-framework/api/pkg/validation/errors"
+	interfaces "github.com/operator-framework/api/pkg/validation/interfaces"
+)
+
+// MultipleArchitecturesValidator validates the bundle against criteria to support Multiple Architectures. For further
+// information check: https://olm.operatorframework.io/docs/advanced-tasks/ship-operator-supporting-multiarch/
+//
+// This validator will inspect the images with the chosen container-tool. One of: [docker, podman, none] (By default docker)
+// and then: (It is only used to $container-tool manifest inspect)
+//
+// - raise a error(s) when is possible to confirm that images does not provide the support defined via to the labels on the CSV
+//
+// - raise a warning when it is possible to check that the Operator manager image(s) supports architecture(s) not defined via labels. Therefore, it shows like the labels are missing.
+//
+// - raise warnings when is possible to verify that the images defined in the CSV does not provide the same architecture(s) supported by the Operator manager image(s) or defined via the labels
+//
+// ### What is checked?
+//
+// On this check, we aggregate the archetype(s) and OS(s) provided via the labels and those which are found by checking the images so that, we can check:
+//
+// - If your CSV is missing labels
+//
+// - If your Operator bundle specifies images which does not supports all archetypes found for your Operator image(s) (probably supported by your project)
+//
+// Note: To better guess the case scenarios where authors might missed the labels the following check will verify all architectures support for the Operator image(s). However, by looking at the CSV we are not able to ensure what is the Operator image because this info is not provided. Therefore, we know by SDK the Operator image container will be called manager.
```

Comment:
```suggestion
// Note: To better guess the case scenarios where authors might have missed the labels, the following check will verify all architectures supported by the Operator image(s). However, by looking at the CSV we are not able to ensure what is the Operator image because this info is not provided. Therefore, we know by SDK the Operator image container will be called manager.
```

Reply:
I'm still not sure what is meant here?